src/main/java/com/softserve/academy/spaced/repetition/controller/CardController.java

```diff
@@ -1,201 +1,100 @@
-    @PreAuthorize(value = "@accessToUrlService.hasAccessToCard(#deckId, #cardId)")
-    public ResponseEntity<CardPublicDTO> getCardByDeck(@PathVariable Long deckId, @PathVariable Long cardId) {
-        Card card = cardService.getCard(cardId);
-        Link selfLink = linkTo(methodOn(CardController.class).getCardByDeck(deckId, cardId)).withSelfRel();
-        CardPublicDTO cardPublicDTO = DTOBuilder.buildDtoForEntity(card, CardPublicDTO.class, selfLink);
-        return new ResponseEntity<>(cardPublicDTO, HttpStatus.OK);
-    }
-
-    @Auditable(action = AuditingAction.CREATE_CARD_VIA_COURSE_AND_DECK)
-    @PostMapping(value = "/api/category/{categoryId}/courses/{courseId}/decks/{deckId}/cards")
-    @PreAuthorize(value = "@accessToUrlService.hasAccessToDeck(#categoryId, #courseId, #deckId)")
-    public ResponseEntity<CardPublicDTO> addCardByCourseAndDeck(@Validated(Request.class) @RequestBody Card card,
-                                                                @PathVariable Long categoryId,
-                                                                @PathVariable Long courseId,
-                                                                @PathVariable Long deckId,
-                                                                @RequestParam List<String> images) {
-
-        cardService.addCard(card, deckId, images);
-        Link selfLink = linkTo(methodOn(CardController.class).getCardByCourseAndDeck(categoryId, courseId, deckId, card.getId())).withSelfRel();
-        CardPublicDTO cardPublicDTO = DTOBuilder.buildDtoForEntity(card, CardPublicDTO.class, selfLink);
-        return new ResponseEntity<>(cardPublicDTO, HttpStatus.CREATED);
+    @GetMapping("not-postponed")
+    @ResponseStatus(HttpStatus.OK)
+    public Boolean areThereNotPostponedCardsAvailable(@PathVariable Long deckId)
+            throws NotAuthorisedUserException {
+        return cardService.areThereNotPostponedCardsAvailable(deckId);
     }
 
     @Auditable(action = AuditingAction.CREATE_CARD_VIA_CATEGORY_AND_DECK)
-    @PostMapping(value = "/api/category/{categoryId}/decks/{deckId}/cards")
-    @PreAuthorize(value = "@accessToUrlService.hasAccessToDeckFromCategory(#categoryId, #deckId)")
-    public ResponseEntity<CardPublicDTO> addCardByCategoryAndDeck(@PathVariable Long categoryId,
-                                                                  @PathVariable Long deckId,
-                                                                  @RequestBody CardDTO card) {
-        LOGGER.debug("Add card to categoryId: {}, deckId: {}", categoryId, deckId);
+    @PostMapping(value = "cards")
+    @ResponseStatus(HttpStatus.CREATED)
+    public CardPublicDTO addCard(@PathVariable Long deckId, @Validated @RequestBody CardDTO card) {
+        LOGGER.debug("Add card to deckId: {}", deckId);
         Card newCard = new Card(card.getTitle(), card.getQuestion(), card.getAnswer());
         cardService.addCard(newCard, deckId, card.getImages());
```

Comment:
Why didn't we get object of `Card` from method `addCard`?

Reply:
i just use created card (newCard), one of ways to write it

Reply:
We can get, for example, another object from method `addCard`, that not be equal to `newCard` object.

Reply:
we need just data of this object